File: scala/dao/interfaces/UserInterface.scala
package dao.interfaces

import entities.User
trait UserInterface {
  def findUserById(id: Long): Option[User]
  def findUser(username: String): Option[User]
  def findAllUser(): Seq[User]
  def addUser(user: User): Boolean
  def updateUser(user: User): Boolean
}

File: lab3/src/main/java/dao/interfaces/UserInterface.java
package dao.interfaces;

import entities.User;
import java.util.List;
import java.util.Optional;

public interface UserInterface {
    Optional<User> findUserById(Long id);
    Optional<User> findUser(String username);
    List<User> findAllUser();
    boolean addUser(User user);
    boolean updateUser(User user);
}
